File: 프로그래머스/3/43164. 여행경로/여행경로.py
from collections import defaultdict, deque
    
def solution(tickets):
    graph = defaultdict(deque)
    path = []
    
    for start, end in tickets:
        graph[start].append(end)
    
    for key in graph:
         graph[key] = deque(sorted(graph[key]))
    
    def dfs(airport):
        while graph[airport] and len(graph[airport]) > 0:
            next = graph[airport].popleft()
            dfs(next)
        path.append(airport)

    dfs('ICN')
    return path[::-1]
            

File: 프로그래머스/3/43164. 여행경로/여행경로.js
function solution(tickets) {
    const graph = {}; 
    const path = [];
    
    for(let i = 0; i < tickets.length; i++){
        const [start, end] = tickets[i];
        graph[start] = (graph[start] || []);
        graph[start].push(end);
    }
    for (const key in graph) {
        graph[key].sort();
    }

    const dfs = (airport) => {
        while(graph[airport] && graph[airport].length){
            const next = graph[airport].shift();
            dfs(next);
        }
         path.push(airport);
    }
    
    dfs('ICN');
    return path.reverse();
}